lib/rspec/matchers/built_in/include.rb

```diff
@@ -153,8 +153,15 @@ def comparing_hash_keys?(expected_item)
         def actual_hash_has_key?(expected_key)
           # We check `key?` first for perf:
           # `key?` is O(1), but `any?` is O(N).
-          actual.key?(expected_key) ||
-          actual.keys.any? { |key| values_match?(expected_key, key) }
+
+          has_exact_key =
+            begin
+              actual.key?(expected_key)
+            rescue
+              false
+            end
```

Comment:
Would the following be more intention-revealing?
```suggestion
          has_exact_key =
            if Matchers.is_a_matcher?(expected_key)
              false
            else
              actual.key?(expected_key)
            end
```

Reply:
No because I want to catch errors in custom implementations, if you matched on any non string key, rack would break, despite any object being a valid hash key.

Reply:
This raises a question if our existing specs are sufficient to cover this. I’ve checked with my suggestion and nothing failed.
So it’s either the ‘is_a_matcher?’ that detects a regexp as a matcher, or the rack-like mock’s ‘key?’ that is not breaking when a non-string (e.g. regexp) is not breaking to mimic how rack does it.